packages/core/src/modules/ledger/__tests__/IndyLedgerService.test.ts

```diff
@@ -59,21 +59,25 @@ describe('IndyLedgerService', () => {
     indyIssuerService = new IndyIssuerServiceMock()
     poolService = new IndyPoolServiceMock()
     const pool = new IndyPool(pools[0], config.agentDependencies, config.logger, new Subject(), new NodeFileSystem())
-    jest.spyOn(pool, 'submitWriteRequest').mockResolvedValue({} as LedgerWriteReplyResponse)
-    jest.spyOn(pool, 'submitReadRequest').mockResolvedValue({} as LedgerReadReplyResponse)
     jest.spyOn(pool, 'connect').mockResolvedValue(0)
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    poolService.ledgerWritePool = pool
+    jest.spyOn(poolService, 'getPoolForNamespace').mockReturnValue(pool)
 
     ledgerService = new IndyLedgerService(config.agentDependencies, config.logger, indyIssuerService, poolService)
   })
 
   describe('LedgerServiceWrite', () => {
     it('should throw an error if the config version does not match', async () => {
+      jest
+        .spyOn(poolService, 'submitWriteRequest')
+        .mockRejectedValue(
+          new Error(
+            'Unable to satisfy matching TAA with mechanism "accept" and version "1.0" in pool.\n Found ["accept"] and version 2.0 in pool.'
+          )
+        )
```

Comment:
So, Can I keep tests in both **IndyLedgerService.test.ts** and **IndyPoolService.test.ts** ?

Reply:
So, Can I keep the same tests in both **IndyLedgerService.test.ts** and **IndyPoolService.test.ts** ?

Reply:
What do you mean exactly? What we're testing here doesn't really test anything, but I do think we should have this test in the indy pool service. Doest hat answer your question?

Reply:
Okay, I will update the test to catch the actual error from the function instead of mocking it and move the tests from IndyLedgerService.test.ts to IndyPoolService.test.ts. Is It right now?

Reply:
Yes if there's no tests left that's fine :+1: